db/version_set.cc

```diff
@@ -1034,7 +1038,7 @@ class LevelIterator final : public InternalIterator {
   bool is_next_read_sequential_;
 };
 
-void LevelIterator::Seek(const Slice& target) {
+void LevelIterator::PassOne(const Slice& target) {
```

Comment:
Like BlockBasedTableIterator, I don't think we need to break it into PassOne and PassTwo. If file_iter_ status after line 1062 is TryAgain(), return. The next seek can simply redo the seek. If the key is the same, need_to_reseek will be set to false and this'll end up directly calling file_iter_.Seek(). It the new seek key is different and falls in a different file, the old file iter destructor will eventually be called, and the destructor should take care of polling/aborting any async IO.